pkg/cortex/cortex.go

```diff
@@ -108,6 +108,17 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
 	flag.IntVar(&chunk_util.QueryParallelism, "querier.query-parallelism", 100, "Max subqueries run in parallel per higher-level query.")
 }
 
+// Validate the cortex config and returns an error if the validation
+// doesn't pass
+func (c *Config) Validate() error {
+	err := c.Schema.Validate()
```

Comment:
I can't see how `Load()` is called before we get here - won't it just be blank?

Reply:
To my understanding, there are three ways to configure the schema config in Cortex:
1. Everything passed as CLI flag
2. Everything passed as CLI flag, but schema config as file via `-config-yaml`
3. The whole config is a yaml via `-config.file`

In the case 3, the schema config is already loaded at this stage and we should validate it here, because the `SchemaConfig.Load()` (called later) will be a no-op (it skips if `len(cfg.Configs) > 0`).

Am I missing anything?

Reply:
I didn’t even know it was possible to put the schema in the same file, ie case 3. 
What happens in case 1? Does that deserve a comment?

Reply:
> What happens in case 1?

In case 1, all flags are parsed by `flag.Parse()` (`main.go`) and the schema config validation is triggered by `SchemaConfig.Load()`.

However, you comment made me realize that my change set doesn't trigger the overall config validation in case of config passed as CLI flags in all cases, so I've pushed a change to fix it (_shouldn't be an issue with the current change set, cause we validate only the schema config, but this PR should be considered the scaffolding for further validations, it's better to have it right first_).

Reply:
Ah, the other thing I was confused about is that it isn't possible to set the period from the CLI, so case 1 doesn't need checking.